test/run-net452-ps5/BasicTugCmdlets.ps1

```diff
@@ -44,27 +44,55 @@ $handlerLogger.LogInformation("  * dscConfigPath  = [$dscConfigPath]")
 $handlerLogger.LogInformation("  * dscModulePath  = [$dscModulePath]")
 
 ## Make sure the paths exist
-mkdir -Force -Path $dscRegKeyPath  | Out-Null
-mkdir -Force -Path $dscRegSavePath | Out-Null
-mkdir -Force -Path $dscConfigPath  | Out-Null
-mkdir -Force -Path $dscModulePath  | Out-Null
+if (!(test-path $dscRegKeyPath)) {
+    new-item -ItemType Directory -Force -Path $dscRegKeyPath
+    }
+if (!(test-path -path $dscRegSavePath)) {
+    new-item -ItemType Directory -Force -Path $dscRegSavePath
+    }
+if (!(test-path -path $dscConfigPath)) {
+    new-item -ItemType Directory -Force -Path $dscConfigPath
+    }
+if (!(Test-Path -Path $dscModulePath)) {
+    new-item -ItemType Directory -Force -Path $dscModulePath
+    }
 
 
 function Register-TugNode {
     param(
         [guid]$AgentId,
         [Tug.Model.RegisterDscAgentRequestBody]$Details
     )
-    $handlerLogger.LogTrace("REGISTER: $($PSBoundParameters | ConvertTo-Json -Depth 3)")
-
+    
     ## Return:
     ##    SUCCESS:  n/a
     ##    FAILURE:  throw an exception
 
+    $handlerLogger.LogTrace("REGISTER: $($PSBoundParameters | ConvertTo-Json -Depth 3)")
+
     $regPath = [System.IO.Path]::Combine($dscRegSavePath, "$($AgentId).json")
     $regInfo = $Details | ConvertTo-Json -Depth 10
-    $handlerLogger.LogDebug("Saving node reg details [$regPath]: $regInfo")
-    Set-Content -Path $regPath -Value $regInfo
+    
+    #New Registration
+    if (!(Test-Path $regPath)) {
+        $handlerLogger.LogDebug("Saving node reg details [$regPath]: $regInfo")
+        Set-Content -Path $regPath -Value $regInfo
+        }
+    
+    #Update existing registration
+    else {
+        set-content -path "$dscRegSavePath\Temp.json" -value $reginfo
```

Comment:
@majst32 -- I would recommend changing from a static `\Temp.json` to a dynamically computed temp file name, like using `New-TemporaryFile` or if you want to keep it local, something like `$tempFile = "$dscRegSavePath\$([Guid]::NewGuid()).json"` so that there is no contention if 2 or more reg update requests from different nodes come in at the same time.